### PR TITLE
Add price filter

### PR DESCRIPTION
**Pull request by akmatchev**
## Overview

Added a price filter so that users can filter posts by a price range.

## Changes Made

Users can put in a lower bound and upper bound for their price range. Then, they can view all of the posts that are within that price range.

## Test Coverage

I tested the feature through manual testing and unit testing. I used the same logic for both forms of testing. First, I tested a case that the price of a post is strictly between the lower bound and the upper bound. Next, I tested the two cases where the price is exactly equal to the lower bound and upper bound. Lastly, I tested the case where the price is not within the price range.

### CODE REVIEW

File: src/repositories/PostRepository.ts [hunk 1]
```diff
@@ -90,6 +91,16 @@ export class PostRepository extends AbstractRepository<PostModel> {
       .getMany();
   }
 
+  public async filterPostsByPrice(lowerBound: number, upperBound: number): Promise<PostModel[]> {
+    return await this.repository
+      .createQueryBuilder("post")
+      .leftJoinAndSelect("post.user", "user")
+      .where("post.price >= :lowerBound", {lowerBound: lowerBound})
+      .andWhere("post.price <= :upperBound", {upperBound: upperBound})
+      .andWhere("post.archive = false")
```
Comment (mateow99) [inline src/repositories/PostRepository.ts:100]: should also have `.andWhere("post.user != user.id)` to eliminate people seeing their own posts.

**Review by mateow99 — APPROVED** [review]
Just address that issue of people seeing own posts across all the post routes. maybe should be its own pr so I'm approving for now.

File: src/repositories/PostRepository.ts [hunk 2]
```diff
@@ -1,3 +1,4 @@
+import internal from 'stream';
```
Comment (snajima) [inline src/repositories/PostRepository.ts:1]: what is this for?

File: src/tests/PostTest.test.ts [hunk 3]
```diff
@@ -1,3 +1,4 @@
+import { Controller } from 'routing-controllers';
```
Comment (snajima) [inline src/tests/PostTest.test.ts:1]: and this?

File: src/tests/PostTest.test.ts [hunk 4]
```diff
@@ -304,6 +305,60 @@ describe('post tests', () => {
...
+    expectedPost.created = getPostsResponse.posts[0].created;
+
+    expect(getPostsResponse.posts).toEqual([expectedPost]);
+
+    filter = {
+      lowerBound: 1.0,
+      upperBound: 2.0,
+    };
+
+    getPostsResponse = await postController.filterPostsByPrice(filter);
```
Comment (snajima) [inline src/tests/PostTest.test.ts:357]: how are these test cases different from each other? if they're not, label this test case as something a little more informative (e.g. `filter posts by price, where price is within range`). if they are, separate them into different test cases.

**Review by snajima — CHANGES_REQUESTED** [review]
great first pr! address the problems i addressed above first :)

**Review by snajima — APPROVED** [review]
lgtm, let's get this merged! also, make sure u hit "re-request review" so i get notified when u address my comments :)

**Pull request merged**